servers/src/main/java/tachyon/master/file/meta/FileSystemMasterView.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.master.file.meta;
+
+import java.util.List;
+
+import com.google.common.base.Preconditions;
+
+import tachyon.exception.FileDoesNotExistException;
+import tachyon.master.file.FileSystemMaster;
+import tachyon.thrift.FileInfo;
+
+/**
+ * This class exposes a read-only view of {@link FileSystemMaster}. This class is thread-safe.
```

Comment:
Why does the lineage master use this? Doesn't it just get passed a FileSystemMaster object?